### ISSUE TITLE
steam-1.0.0.69 404 error when installing

### ISSUE THREAD

**Issue by kfajdsl** [NixOS/nixpkgs#120539]
**Describe the bug**
The `steam` package fails to install, giving a 404 error. I'm using nixos-unstable. 

**To Reproduce**
1. `nix-env -iA nixos.steam`

**Expected behavior**
Steam successfully installs.

**Actual behavior**
The following output is given:
```
...
building '/nix/store/12l3lkzsihr61066v6z9m71ghapj4b4f-steam_1.0.0.69.tar.gz.drv'...
building '/nix/store/0wbm7i851ywca3xk3xh0kdrza26cvkmm-steam-runtime-0.20201203.1.drv'...
building '/nix/store/i13s2dq882k0c02ibxlys2l2wch1ibj3-steam-runtime-0.20201203.1.drv'...

trying https://repo.steampowered.com/steam/pool/steam/s/steam/steam_1.0.0.69.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (22) The requested URL returned error: 404
error: cannot download steam_1.0.0.69.tar.gz from any mirror
builder for '/nix/store/12l3lkzsihr61066v6z9m71ghapj4b4f-steam_1.0.0.69.tar.gz.drv' failed with exit code 1
cannot build derivation '/nix/store/8md8808rprpj5zylhc47zlwqn4bs1zj9-steam-original-1.0.0.69.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/a24ngp4grqmlgmds6wcdk6vvznc8qg1f-steam.drv': 1 dependencies couldn't be built
error: build of '/nix/store/a24ngp4grqmlgmds6wcdk6vvznc8qg1f-steam.drv' failed
```
**Additional context**
https://repo.steampowered.com/steam/pool/steam/s/steam/ seems to only have `steam_1.0.0.61.tar.gz` and `steam_1.0.0.70.tar.gz`.

Maybe switch to https://repo.steampowered.com/steam/archive/stable/steam_1.0.0.69.tar.gz? This is taken from https://repo.steampowered.com/steam/:

> The apt package pool in pool/ only contains the newest stable release and the newest beta. Use the version-numbered releases in archive/stable/ or archive/beta/ if you need a stable URL (in Arch PKGBUILDs, Gentoo ebuilds, etc.) or if you need to scan a directory listing to watch for new releases (in debian/watch etc.).

**Notify maintainers**

@jagajaga
@jonringer 

**Metadata**
 - system: `"x86_64-linux"`
 - host os: `Linux 5.11.14, NixOS, 21.05pre283883.d235056d6d6 (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.10`
 - channels(sahan): `"home-manager, nixos-unstable-21.05pre283883.d235056d6d6"`
 - channels(root): `"nixos-21.05pre283883.d235056d6d6"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

Maintainer information:
```yaml
attribute:
steam
module:
programs.steam
```

**Comment by jonringer** [comment]
fixed in #120545

**Comment by jonringer** [comment]
thank you @kfajdsl  for opening up a ticket
